packages/features/ee/teams/components/TeamEventTypeForm.tsx

```diff
@@ -0,0 +1,144 @@
+    } else {
+      form.setValue("metadata", null);
+    }
+  }, [schedulingTypeWatch]);
+
+  const { register } = form;
+
+  const createMutation = trpc.viewer.eventTypes.create.useMutation({
+    onSuccess: async () => {
+      await utils.viewer.eventTypes.getByViewer.invalidate();
+      router.push(`/settings/teams/${teamId}/profile`);
+      form.reset();
+    },
+  });
+
+  const urlPrefix = orgBranding?.fullDomain ?? process.env.NEXT_PUBLIC_WEBSITE_URL;
+
+  return (
+    <>
+      <Form
+        form={form}
+        handleSubmit={(values) => {
+          createMutation.mutate(values);
+        }}>
+        <div className="mt-3 space-y-6 pb-11">
+          <TextField
+            type="hidden"
+            labelProps={{ style: { display: "none" } }}
+            {...register("teamId", { valueAsNumber: true })}
+            value={teamId}
+          />
+          <TextField
+            label={t("title")}
+            placeholder={t("quick_chat")}
+            data-testid="event-type-quick-chat"
+            {...register("title")}
+            onChange={(e) => {
+              form.setValue("title", e?.target.value);
+              if (form.formState.touchedFields["slug"] === undefined) {
+                form.setValue("slug", slugify(e?.target.value));
+              }
+            }}
+          />
+
+          {urlPrefix && urlPrefix.length >= 21 ? (
+            <div>
+              <TextField
+                label={`${t("url")}: ${urlPrefix}`}
+                required
+                addOnLeading={
+                  <Tooltip content={!isManagedEventType ? `team/${team?.slug}` : t("username_placeholder")}>
+                    <span className="max-w-24 md:max-w-56">
+                      /{!isManagedEventType ? `team/${team?.slug}` : t("username_placeholder")}/
+                    </span>
+                  </Tooltip>
+                }
+                {...register("slug")}
+                onChange={(e) => {
+                  form.setValue("slug", slugify(e?.target.value), { shouldTouch: true });
+                }}
+              />
+
+              {isManagedEventType && (
+                <p className="mt-2 text-sm text-gray-600">{t("managed_event_url_clarification")}</p>
+              )}
+            </div>
+          ) : (
+            <div>
+              <TextField
+                label={t("url")}
+                required
+                addOnLeading={
+                  <Tooltip
+                    content={`${urlPrefix}/${
+                      !isManagedEventType ? `team/${team?.slug}` : t("username_placeholder")
+                    }/`}>
+                    <span className="max-w-24 md:max-w-56">
+                      {urlPrefix}/{!isManagedEventType ? `team/${team?.slug}` : t("username_placeholder")}/
+                    </span>
+                  </Tooltip>
+                }
+                {...register("slug")}
+              />
```

Comment:
It would be great to have this handled in a separate component to improve semanticity and readability of the Form, the component can be in the same file

Reply:
Hey @ThyMinimalDev, we're moving the entire form to a separate location, @hariombalhara also suggested that.